log/logger.go

```diff
@@ -151,5 +145,33 @@ func SetCtxLogger(ctx context.Context, key string, value interface{}) (context.C
 	var lggr logrus.Logger
 	newLogEntry := &StructuredLoggerEntry{Logger: lggr.WithField(key, value)}
 	nCtx := context.WithValue(ctx, CtxLoggerKey, newLogEntry)
+
 	return nCtx, newLogEntry.Logger
 }
+
+func ErrorExtra(ctx context.Context, msg string, fields logrus.Fields) (context.Context, logrus.FieldLogger) {
+	logger := GetCtxLogger(ctx)
+	logger.WithFields(fields)
+	logger.Error(msg)
+
+	nCtx := context.WithValue(ctx, CtxLoggerKey, &StructuredLoggerEntry{Logger: logger})
+	return nCtx, logger
+}
+
+func WarnExtra(ctx context.Context, msg string, fields logrus.Fields) (context.Context, logrus.FieldLogger) {
+	logger := GetCtxLogger(ctx)
+	logger.WithFields(fields)
```

Comment:
I suspect the `WithFields` method works a little differently -- you might need to capture its return value, which should be a new entry with the extra fields added. I think a unit test might help clear up whether I've got this wrong, though.